homeassistant/components/lock/__init__.py

```diff
@@ -97,8 +100,11 @@ async def _async_lock(entity: LockEntity, service_call: ServiceCall) -> None:
 
 async def _async_unlock(entity: LockEntity, service_call: ServiceCall) -> None:
     """Unlock the lock."""
-    code: str = service_call.data.get(ATTR_CODE, "")
-    if entity.code_format_cmp and not entity.code_format_cmp.match(code):
+    if (
+        (code := service_call.data.get(ATTR_CODE))
```

Comment:
If we go this route, lets at least not accept an empty string.

```suggestion
        (code := service_call.data.get(ATTR_CODE)) is not None
```